sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1252,7 +1252,9 @@ class Analyzer(
           case ae: AnalysisException => filter
         }
 
-      case sort @ Sort(sortOrder, global, aggregate: Aggregate) if aggregate.resolved =>
+      case sort @ Sort(sortOrder, global, aggregate: Aggregate) if aggregate.resolved &&
+          (conf.orderByOrdinal && sortOrder.forall(x => IntegerIndex.unapply(x.child).isEmpty)) =>
```

Comment:
I have the feeling that this guard is wrong. This disables this entire clause if `conf.orderByOrdinal` is `false`. Shouldn't it be: `!conf.orderByOrdinal || sortOrder.forall(x => IntegerIndex.unapply(x.child).isEmpty)`